AI content updates (#17994)
ethereum/ethereum-org-website: public/content/ai-agents/getting-started/index.md

---
title: Getting started with AI agents
description: A developer on-ramp for building AI agents on Ethereum, from wallet creation to framework connection
lang: en
template: ai-agents
emoji: ":rocket:"
sidebarDepth: 2
image: /images/ai-agents/hero-image.png
alt: Getting started with AI agents on Ethereum

> summaryPoint1: Deploy a smart account wallet, fund it, and connect an agent framework in four steps

Reviewer: Trim to ≤8 words; keep keywords; parallel structure.

```suggestion
summaryPoint1: Set up your first agent in four steps
```

> summaryPoint1: Deploy a smart account wallet, fund it, and connect an agent framework in four steps

Reviewer: Trim to ≤8 words; keep keywords; parallel structure.

```suggestion
summaryPoint1: Set up your first agent in four steps
```

> summaryPoint2: Configure session key spending policies to give your agent safe, bounded autonomy

Reviewer: Trim to ≤8 words; keep keywords; parallel structure.

```suggestion
summaryPoint2: Configure session keys for bounded agent autonomy
```

> summaryPoint2: Configure session key spending policies to give your agent safe, bounded autonomy

Reviewer: Trim to ≤8 words; keep keywords; parallel structure.

```suggestion
summaryPoint2: Configure session keys for bounded agent autonomy
```

> summaryPoint3: Working code examples for smart account setup, framework connection, and spending guardrails

Reviewer: Trim to ≤8 words; keep keywords; parallel structure.

```suggestion
summaryPoint3: Code examples for setup, integration, and guardrails
```

> summaryPoint3: Working code examples for smart account setup, framework connection, and spending guardrails

Reviewer: Trim to ≤8 words; keep keywords; parallel structure.

```suggestion
summaryPoint3: Code examples for setup, integration, and guardrails
```

Before you start, have the following ready.

**A funded development environment.** You need [Node.js (v18+)](https://nodejs.org) and a package manager (npm, yarn, or pnpm). If you are new to Ethereum development, read the resources in [the Developers hub](/developers/) for context and resources about building on the network, and the [Ethereum for AI agents](/ai-agents/ethereum/) hub page for the specific capabilities your agent will use.

**An LLM API key.** Your agent framework needs access to a language model to drive its reasoning loop. The examples on this page use the OpenAI API, but any provider supported by the Vercel AI SDK (Anthropic, Google, Mistral, and others) works the same way. Set your key in your environment before running any code.

**Infrastructure API keys.** Account abstraction requires a bundler (to relay UserOperations) and optionally a Paymaster (to sponsor gas). [ZeroDev](https://zerodev.app/), [Pimlico](https://www.pimlico.io/), and [Alchemy](https://www.alchemy.com/) all offer free testnet tiers. Create a project and collect your RPC endpoints now.

Create a `.env` file at the root of your project and populate it before running any of the examples below:

> ```bash

Reviewer: Example of the issue: this ```bash``` block, and every code fence on this page, renders as a plain styled `<pre>` on the deploy preview, not through the site's `Codeblock` component (no Prism highlighting, no copy button, no line numbers, no language badge). Same thing on `frameworks/` (line 59), `wallets/` (line 116), `identity/` (line 90), and `verification/` (line 50). Root cause and proposed 4-line fix in `src/layouts/Static.tsx` are in the main review comment.

Author: sg, addressed with commit 8000431
